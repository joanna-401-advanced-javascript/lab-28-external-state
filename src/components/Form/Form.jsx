import React from 'react';

export default class extends React.Component {
  handleChange = (event) => {
    event.preventDefault();
    this.props.handleName(event.target.value);
  };

  handleSubmit = (event) => {
    event.preventDefault();
    this.props.handleCount();
  };

  render(){
    return(
      <form onSubmit={this.handleSubmit}>
        <input type="text" onChange={this.handleChange}/>
      </form>
    );
  }
}
