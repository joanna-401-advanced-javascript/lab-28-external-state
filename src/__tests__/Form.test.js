import React from 'react';
import renderer from 'react-test-renderer';
import App from '../app';
import Form from '../components/Form/Form';

describe('<Form />', () => {
  it('nothing from this page is rendered at application start', () => {
    const form = shallow(<Form />);
    expect(form.find('form').exists()).toBe(true);
    expect(form.find('input').exists()).toBe(true);
  });

  it('can change app state name on change', () => {
    const app = mount(<App />);
    const input = app.find('input');
    input.simulate('change', { target: { value: 'Jo' } });
    expect(app.state('name')).toEqual('Jo');
  });

  it('can change app state count on change', () => {
    const app = mount(<App />);
    const submit = app.find('form');
    submit.simulate('submit');
    expect(app.state('count')).toEqual(1);
  });

  it('snapshot is rendered correctly', () => {
    const form = renderer
      .create(<Form />)
      .toJSON();
    expect(form).toMatchSnapshot();
  });
});
