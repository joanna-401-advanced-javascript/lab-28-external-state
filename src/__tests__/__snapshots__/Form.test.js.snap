// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> snapshot is rendered correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
  />
</form>
`;
